src/components/settings-menu/DevSection.js

```diff
@@ -119,6 +123,47 @@ const DevSection = () => {
     setErrorObj({ error: 'this throws render error' });
   };
 
+  const handleSubscription = action => {
+    Object.keys(wallets).forEach(async key => {
+      const wallet = wallets[key];
+      const address = wallet.addresses[0].address;
```

Comment:
Do you know what wallet represents in this case? Do HD wallets / child addresses get consolidated into wallet or are they spread out?

Is this the breakdown if I watch an address and have my own wallet with 8 addresses?
- wallet[key1] = watched address
  - wallet.addresses.length = 1
- wallet[key2] = HD wallet
  - wallet.addresses.length = 8

If this is what the breakdown looks like then we need to make sure to loop through all of the wallet addresses and not just pick the first one

Reply:
@welps What are HD wallets / child addresses? Got an example I could plug into the app? I don't have deep knowledge about the blockchain yet.

Only tested this with regular mainnet wallets which have a single address. Seen that method of grabbing it in other parts of the app.

Data dump of `wallets` [here](https://gist.github.com/pugson/214aad9d61eb5d164869199234560e65).

Reply:
I agree with @welps I guess and sent you an explanation in DM